src/Raven.Server/Documents/Indexes/Persistence/Lucene/Documents/AnonymousLuceneDocumentConverter.cs

```diff
@@ -77,7 +77,7 @@ protected override int GetFields<T>(T instance, LazyStringValue key, object docu
 
             if (_reduceOutput)
             {
-                instance.Add(GetReduceResultValueField(Scope.CreateJson(reduceResult, indexContext)));
+                instance.Add(GetReduceResultValueField(Scope.CreateJson(reduceResult, indexContext), writeBuffer));
```

Comment:
Here we are doing something that may be very bad. We are passing the whole buffer here, but it may be larger than needed.

Reply:
Which is fine? We did exactly the same thing before. Check `GetReduceResultValueField` we are getting there a buffer, but to Lucene we are passing the 'reduceResult.Size'

Reply:
No, not fine.
Previously, we created a new instance, which was guaranteed to be zero.
So if we had more data, it was just zero all the time.
Now, you have a shared buffer, which may be dirty, so the data may be *different* on the portions you didn't use.

Reply:
But since we are copying from the beginning of the buffer till the reduceResult.Size in `GetReduceResult` method and then passing the reduceResult.Size to Lucene, so it should be fine? Previously the buffer was for the whole index lifetime of an index, so after few runs it was 'dirty' too?

Reply:
I'm sorry, you are absolutely correct, I missed that part.